base/libdl.jl

```diff
@@ -205,6 +205,43 @@ if is_linux()
     end
 end # linux-only
 
+if Sys.KERNEL == :FreeBSD
```

Comment:
The behavior is same on NetBSD and Dragonfly. But OpenBSD will get compilation error with
`#include <link.h>`. (`link.h` is described on OpenBSD's manual, seems there are some syntax error in the header file)

I decide to use `f is_bsd() && !is_apple()`.